features/step_definitions/workflow.js

```diff
@@ -0,0 +1,208 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const github = require('../support/github');
+const request = require('../support/request');
+const sdapi = require('../support/sdapi');
+const { defineSupportCode } = require('cucumber');
+
+const TIMEOUT = 240 * 1000;
+
+defineSupportCode(({ Before, Given, When, Then }) => {
+    Before({
+        tags: '@workflow'
+    }, function hook() {
+        this.repoOrg = this.testOrg;
+        this.repoName = 'functional-workflow';
+        this.pipelineId = null;
+        this.builds = null;
+    });
+
+    Given(/^an existing pipeline on "(.*)" branch with the workflow jobs:$/, {
+        timeout: TIMEOUT
+    }, function step(branch, table) {
+        return this.getJwt(this.apiToken)
+            .then((response) => {
+                this.jwt = response.body.token;
+
+                return github.createBranch(this.gitToken, branch, this.repoOrg, this.repoName);
+            })
+            .then(() => this.createPipeline(this.repoName, branch))
```

Comment:
Do we need recreating a pipeline like other test cases?

Reply:
I thought that we didn't have to recreate pipelines in workflow tests, but I will lessen codes by  using `ensurePipelineExists`.